doc/auto_techsupport_and_coredump_mgmt.md

```diff
@@ -462,96 +474,25 @@ Enhance the existing techsupport sonic-mgmt test with the following cases.
 |  2   | Check if the techsupport cleanup is working as expected                                                                                 |
 |  3   | Check if the global rate-& & per-process rate-limit-interval is working as expected                                                     |
 |  4   | Check if the core-dump cleanup is working as expected                                                                                   |
-|  5   | Check if the core-dump generated when reaching memory threshold                                                                                   |
-## 8. SONiC-to-SONiC Upgrade Considerations
+|  5   | Check if the core-dump generated when reaching memory threshold                                                                         |
 
-The default config required for auto_techsupport is present in the init_cfg.json. Therefore, when a clean installation of SONiC is performed, the configuration is found in the config DB and the feature is active. 
+## 9. SONiC-to-SONiC Upgrade Considerations
 
-However, in the case of SONiC-SONiC upgrade, the previous config_db.json is migrated and init_cfg.json is not involved. In that case, it is the responsibility of the admin to provide the config, if the admin wants to leverage this feature. 
+The configuration in the init_cfg.json is loaded to the running config i.e. CONFIG_DB even in the case of SONiC-SONiC upgrade from a older image which doesn't support this feature.
 
-Load this Example config provided below to enable the feature. Each of the fields are explained in Section 4 and can be modified accordingly
+### 10  App Extension Considerations
+
+Detailed Info related to Appliation Extension can be found here: https://github.com/Azure/SONiC/blob/master/doc/sonic-application-extension/sonic-application-extention-hld.md
+
+A new AUTO_TECHSUPPORT_FEATURE register/deregister option will be introduced. The existing FeatureRegistry class will be enahcned to also add/delete configuration related to AUTO_TECHSUPPORT_FEATURE table.
+
+This will be run when the application installs/uninstalls. Since, the auto-techsupport feature uses compile time flag to determine whether to enable/disable itself, it is not possible to determine that at runtime when the application is installed. 
+
+Thus the decision to whether or not to enable the new feature will be based on the current values of AUTO_TECHSUPPORT & AUTO_TECHSUPPORT_FEATURE tables. The default value for new feature will be disabled if the global state is shown disabled in init_cfg.json. If not, the feature will be enabled. The rate-limit-interval & memory threshold is set to 600 & 10% by default. 
+
+If there is a user-provided config in conifg_db.json or CONFIG_DB, it is not overwritten by the default config when the app installs and is not cleared when the app gets uninstalled.

```

Comment:
Wonder what is the motivation? In the current design of sonic-package-manager the default behaviour is to delete all configuration related to the feature. User may choose to leave it if he uninstalls with "--keep-config". https://github.com/Azure/sonic-utilities/blob/master/sonic_package_manager/main.py#L417

Reply:
Only the default configuration is cleared when uninstall is invoked. If "--keep-config" is used (any user provided config + default config) stays. 

The idea is not to delete/overwrite any user provided config. 

Let's say user want to use different rate_limit_interval or available_mem_threshold than the default values, with this approach he can load the config anytime (i.e. before of after installation and can be sure that this is the configuration which is used)

Reply:
All configuration is removed when keep_config is False, including user config, not just the default. This is by design.
https://github.com/Azure/sonic-utilities/blob/master/sonic_package_manager/service_creator/creator.py#L523

Reply:
Okay, i'll follow this then

Reply:
Handled. Please review